### ISSUE TITLE
Feature request: Findassimp.cmake

### ISSUE THREAD

**Issue by claudiofantacci** [robotology/ycm#224]
Following https://github.com/robotology/superimpose-mesh-lib/issues/6, I propose to add `Findassimp.cmake` in YCM.
To give you an example, [here are](https://github.com/robotology/superimpose-mesh-lib/blob/ee1acab95f15a5d67b8b05a56378ed5d729b0345/src/SuperimposeMesh/CMakeLists.txt#L28-L71) the workaround that I have in `superimpose-mesh-lib`. Basically a `Findassimp` itself.

Please read [this comment](https://github.com/robotology/superimpose-mesh-lib/issues/6#issuecomment-473279341) to better understand the problem.
As suggested by @traversaro, we could add [dartsim's `Findassimp.cmake`](https://github.com/dartsim/dart/blob/master/cmake/Findassimp.cmake).
This would really help a lot.

What do you think?

**Comment by drdanz** [comment]
I don't know, but overriding the config file from upstream is not really recommended, and it will cause complications when upstream is fixed.
I recommend to fix it upstream and in the meanwhile add a module that exports the exact same variables and targets as the config file upstream, so that, as soon as we remove it, there won't be any need of a transition.

**Comment by claudiofantacci** [comment]
> I recommend to fix it upstream

Upstream is fixed, but not yet released officially.



> in the meanwhile add a module that exports the exact same variables and targets as the config file upstream, so that, as soon as we remove it, there won't be any need of a transition.

I'm working on it 👍

**Comment by claudiofantacci** [comment]
I implemented the first version of the `Findassimp.cmake`.
It needs to be plished a bit and we need to discuss few things.

To start off, the upstream config module is quite complicated.
Long story short, the config immediately and only calls [assimpTargets.cmake](https://github.com/assimp/assimp/blob/master/assimpTargets.cmake.in) that then calls [assimpTargets-Relese/Debug](https://github.com/assimp/assimp/blob/master/assimpTargets-release.cmake.in).

By my understanding these are the variables that the config exports:
```
ASSIMP_ROOT_DIR - the root directory where the installation can be found
ASSIMP_CXX_FLAGS - extra flags for compilation
ASSIMP_LINK_FLAGS - extra flags for linking
ASSIMP_INCLUDE_DIRS - include directories
ASSIMP_LIBRARY_DIRS - link directories
ASSIMP_LIBRARIES - libraries to link plugins with
ASSIMP_Boost_VERSION - the boost version assimp was compiled with
```
as well as the `assimp::assimp` target.

[Here](https://github.com/claudiofantacci/ycm/blob/fa1276f27d7804932678458ed9220ea05a5db4cd/find-modules/Findassimp.cmake) is the first version of the `Findassimp.cmake` module that I implemented.

It has the most important variables and the target and it lacks:
 - `ASSIMP_ROOT_DIR`: do you think is important?
 - `ASSIMP_LINK_FLAGS`, `ASSIMP_CXX_FLAGS` and `ASSIMP_Boost_VERSION`: I don't think I can replicate this, unless I call the module first, if present, discard almost everithing because is most likely to be broken and finally get this variables. For now, I simply ingored them. (in my vcpkg and local installations they are empty...)
 - `ASSIMP_LIBRARY_DIRS`: can do this.
```

Let me know what you think and I will refine it.

**Comment by claudiofantacci** [comment]
Implemented new version with `ASSIMP_ROOT_DIR`, `ASSIMP_LIBRARY_DIRS` and `ASSIMP_VERSION` [here](https://github.com/claudiofantacci/ycm/blob/impl/findassimp/find-modules/Findassimp.cmake).

**Comment by claudiofantacci** [comment]
I'm going to refine the above-mentioned implementation and open a PR for this.

**Comment by drdanz** [comment]
Fixed by #229
